src/hooks/Transactor.ts

```diff
@@ -82,90 +94,61 @@ export function useTransactor({
         options?.onDone?.()
         return false
       }
-
       if (!signer || !chain) {
         options?.onDone?.()
         return false
       }
 
-      const tx: Deferrable<TransactionRequest> =
-        options?.value !== undefined
-          ? contract[functionName](...args, { value: options.value })
-          : contract[functionName](...args)
-
-      const reportArgs = Object.values(contract.interface.functions)
-        .find(f => f.name === functionName)
-        ?.inputs.reduce(
-          (acc, input, i) => ({
-            ...acc,
-            [input.name]: args[i],
-          }),
-          {},
-        )
+      logTx({ functionName, contract, args })
 
       if (process.env.NODE_ENV === 'development') {
         await simulateTransaction({ contract, functionName, args, userAddress })
       }
 
-      console.info(
-        '🧃 Transactor::Calling ' + functionName + '() with args:',
-        reportArgs,
-        tx,
-      )
-
-      const txTitle = options?.title ?? functionName
-
       try {
-        let result: unknown
-
-        if (tx instanceof Promise) {
-          console.info('Transactor::AWAITING TX', tx)
-          result = await tx
+        const tx = prepareTransaction({ functionName, contract, args, options })
+        const result: TransactionResponse = await tx
 
-          addTransaction?.(txTitle, result as TransactionResponse, {
-            onConfirmed: options?.onConfirmed,
-            onCancelled: options?.onCancelled,
-          })
-        } else {
-          console.info('Transactor::RUNNING TX', tx)
-
-          if (!tx.gasPrice) tx.gasPrice = gasPrice ?? parseUnits('4.1', 'gwei')
-          if (!tx.gasLimit) tx.gasLimit = hexlify(120000)
+        console.info('✅ Transactor::submitted', result)
 
-          result = await signer.sendTransaction(tx)
-          const txResponse = result as TransactionResponse
-
-          addTransaction?.(txTitle, txResponse, {
-            onConfirmed: options?.onConfirmed,
-            onCancelled: options?.onCancelled,
-          })
```

Comment:
this code is never actually used. It's only used when you have a local eth chain set up, which none of us ever do.

I think its safe to remove simply for the sake of maintainability of this file.

Reply:
Ok sounds good